sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/test/scala/com/azure/cosmos/spark/CosmosClientConfigurationSpec.scala

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.ConsistencyLevel
+
+class CosmosClientConfigurationSpec extends UnitSpec {
```

Comment:
all the other config tests parsing are in `CosmosConfigSpec`. I wonder if this can be merged to same test class?
once you move the parsing capability from `CosmosClientConfiguration` to `CosmosConfig` (as explained in earlier comment) you will be able to rely on ``CosmosConfigSpec` tests

Reply:
This is not parsing config (Map[string, string]), this is the Spec for the ComsosClientConfiguration case class

Reply:
It needs the config Map but internally it uses the other parser to read it and construct the case class values